src/main/client_context.cpp

```diff
@@ -213,10 +214,9 @@ unique_ptr<QueryResult> ClientContext::ExecutePreparedStatement(ClientContextLoc
 
 	bool create_stream_result = statement.allow_stream_result && allow_stream_result;
 	if (enable_progress_bar) {
-		if (progress_bar) {
-			progress_bar.reset();
+		if (!progress_bar) {
+			progress_bar = make_shared<ProgressBar>(&executor, wait_time);
```

Comment:
@pdet what was the rationale to create a progress bar instance every time a query is executed?

Reply:
I don't remember any specific reason for it, it seems that it should be able for us to have only one and reuse it.